glue-codes/openfast-cpp/src/FAST_Prog.cpp

```diff
@@ -38,31 +57,12 @@ void readTurbineData(int iTurb, fast::fastInputs & fi, YAML::Node turbNode) {
   } else {
       fi.globTurbineData[iTurb].TurbineHubPos =  std::vector<double>(3,0.0);
   }
-  if (turbNode["num_force_pts_blade"]) {
-      fi.globTurbineData[iTurb].numForcePtsBlade = turbNode["num_force_pts_blade"].as<int>();
-  } else {
-      fi.globTurbineData[iTurb].numForcePtsBlade = 0;
-  }
-  if (turbNode["num_force_pts_tower"]) {
-      fi.globTurbineData[iTurb].numForcePtsTwr = turbNode["num_force_pts_tower"].as<int>();
-  } else {
-      fi.globTurbineData[iTurb].numForcePtsTwr = 0;
-  }
-  if (turbNode["nacelle_cd"]) {
-      fi.globTurbineData[iTurb].nacelle_cd = turbNode["nacelle_cd"].as<float>();
-  } else {
-      fi.globTurbineData[iTurb].nacelle_cd = 0.0;
-  }
-  if (turbNode["nacelle_area"]) {
-      fi.globTurbineData[iTurb].nacelle_area = turbNode["nacelle_area"].as<float>();
-  } else {
-      fi.globTurbineData[iTurb].nacelle_area = 0.0;
-  }
-  if (turbNode["air_density"]) {
-      fi.globTurbineData[iTurb].air_density = turbNode["air_density"].as<float>();
-  } else {
-      fi.globTurbineData[iTurb].air_density = 0.0;
-  }
+  get_if_present(turbNode, "num_force_pts_blade", fi.globTurbineData[iTurb].numForcePtsBlade, 0);
+  get_if_present(turbNode, "num_force_pts_tower", fi.globTurbineData[iTurb].numForcePtsTwr, 0);
+  float fZero = 0.0;
```

Comment:
No need for default, since that is the default parameter in the function definition. 

I think these parameters should be `get_required`?

Reply:
I think these parameters should be `get_required` when reading in an input file through a CFD solver for actuator line simulations. 

`FAST_Prog.cpp` is only provided as an example and not intended to be a production level glue code using the C++ API. The executable generated using `FAST_Prog.cpp` takes no arguments and expects an input file specifically named `cDriver.i`. If we decide to make `FAST_Prog.cpp` a more production level glue-code, I think it'll require a few more changes.